File: DependencyInjection/ServiceLocatorDIAntipattern/_2_ServiceLocator/Outbound/TcpSocket.cs
using System;

namespace ServiceLocatorDIAntipattern._2_ServiceLocator.Outbound
{
  internal interface IOutputSocket
  {
    void Open();
    void Close();
    void Send(string lol);
  }

  class TcpSocket : IOutputSocket
  {
    public void Open()
    {
      Console.WriteLine("open");
    }

    public void Close()
    {
      Console.WriteLine("closing");
    }

    public void Send(string lol)
    {
      Console.WriteLine(lol);
    }
  }
}

File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/_4_BastardInjection/Outbound/TcpSocket.java
package com.github.grzesiek_galezowski.BastardInjection._4_BastardInjection.Outbound;

interface IOutputSocket
  {
    void Open();
    void Close();
    void Send(String lol);
  }

  class TcpSocket implements IOutputSocket {
    public void Open() {
      System.out.println("open");
    }

    public void Close() {
      System.out.println("closing");
    }

    public void Send(String lol) {
      System.out.println(lol);
    }
  }
